curvefs/src/metaserver/partition_cleaner.h

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * @Project: curve
+ * @Date: 2021-12-15 10:53:45
+ * @Author: chenwei
+ */
+
+#ifndef CURVEFS_SRC_METASERVER_PARTITION_CLEANER_H_
+#define CURVEFS_SRC_METASERVER_PARTITION_CLEANER_H_
+
+#include <memory>
+#include "curvefs/src/metaserver/copyset/copyset_node_manager.h"
+#include "curvefs/src/metaserver/partition.h"
+#include "curvefs/src/metaserver/s3/metaserver_s3_adaptor.h"
+
+namespace curvefs {
+namespace metaserver {
+class PartitionCleaner {
```

Comment:
Need some metric here to know how many deletion tasks are in progress.

Reply:
done
add metric "partition_clean_manager_cleaner"